bodhi/server/consumers/masher.py

```diff
@@ -1006,3 +1037,24 @@ def run(self):
         else:
             self.success = True
         return out, err, returncode
+
+def system(cmd):
+    """
+    Invoke a shell command.
+    :returns: A tuple of output, err message, and return code
+    """
+    ret = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True)
+    out, err = ret.communicate()
+    return out, err, ret.returncode
+
+
+def git_checkout(dirpath, url, branch):
+    """Checks out the given url in the directory and given branch    
+    :param dirpath: Path to checkout  
+    :param url: URL to clone
+    :param branch: The branch to checkout
+    :return: None
+    """
+    cmd = "git clone -b %s %s %s" % (branch, url, dirpath)
+    system(cmd)
```

Comment:
This is OK, but probably not that necessary. ```git clone``` is already used elsewhere in masher as a one-liner so I'm not sure it's that beneficial to have another function do it. That said, if you really want such a function I'm fine with it, but please refactor the other use of it to also use this function if you do so. Optional.